[BUG] v3 installation fails (autocomplete)

### Describe the bug.

`$ yarn global add @asyncapi/cli`

```
[4/4] Building fresh packages...
error /home/runner/.config/yarn/global/node_modules/@asyncapi/cli: Command failed.
info Visit https://yarnpkg.com/en/docs/cli/global for documentation about this command.
Exit code: 1
Command: node ./scripts/enableAutoComplete.js
Arguments: 
Directory: /home/runner/.config/yarn/global/node_modules/@asyncapi/cli
Output:
node:internal/modules/cjs/loader:1215
  throw err;
  ^

Error: Cannot find module '/home/runner/.config/yarn/global/node_modules/@asyncapi/cli/scripts/enableAutoComplete.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1212:15)
    at Module._load (node:internal/modules/cjs/loader:1043:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:1[64](https://github.com/RobinTail/zod-sockets/actions/runs/14273003801/job/40010274524?pr=438#step:6:65):12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v20.19.0
```

### Expected behavior

Should not fail

### Screenshots

<img width="1345" alt="Image" src="https://github.com/user-attachments/assets/1885d417-5679-43d1-b054-c7e24864da12" />

### How to Reproduce

https://github.com/RobinTail/zod-sockets/blob/a23934eaa0e04966677a061fbe22c0c5baa5eacd/.github/workflows/async-api-validation.yml

### 🖥️ Device Information [optional]

- Operating System (OS): Ubuntu  24.04.2  LTS
- Browser: none
- Browser Version: none


### 👀 Have you checked for similar open issues?

- [x] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [x] I have read the [Contributing Guidelines](https://github.com/asyncapi/.github/blob/master/CONTRIBUTING.md)

### Are you willing to work on this issue ?

No, someone else can work on it

--- a/.changeset/curly-cars-knock.md
+++ b/.changeset/curly-cars-knock.md
@@ -1,5 +1,6 @@
 ---
-"@fake-scope/fake-pkg": patch
+'@asyncapi/cli': patch
 ---
 
-Fixing script detection issue in version 3.0.0
+- Fixes script detection issue in version 3.0.0
+- Fixes testing by testing the github action with local CLI

--- a/.github/workflows/test-action.yml
+++ b/.github/workflows/test-action.yml
@@ -75,6 +75,8 @@ jobs:
           docker load --input /tmp/asyncapi.tar
           docker image ls -a
       - uses: actions/checkout@v4
+      - name: Remove bin directory to disable developer mode
+        run: rm -rf ./bin
       - name: Test GitHub Action
         uses: ./
         with:
@@ -107,6 +109,8 @@ jobs:
           docker load --input /tmp/asyncapi.tar
           docker image ls -a
       - uses: actions/checkout@v4
+      - name: Remove bin directory to disable developer mode
+        run: rm -rf ./bin
       - name: Test GitHub Action
         uses: ./
         with:
@@ -128,6 +132,8 @@ jobs:
           docker load --input /tmp/asyncapi.tar
           docker image ls -a
       - uses: actions/checkout@v4
+      - name: Remove bin directory to disable developer mode
+        run: rm -rf ./bin
       - name: Test GitHub Action
         uses: ./
         with:
@@ -162,6 +168,8 @@ jobs:
           docker load --input /tmp/asyncapi.tar
           docker image ls -a
       - uses: actions/checkout@v4
+      - name: Remove bin directory to disable developer mode
+        run: rm -rf ./bin
       - name: Test GitHub Action
         uses: ./
         with:
@@ -195,6 +203,8 @@ jobs:
           docker load --input /tmp/asyncapi.tar
           docker image ls -a
       - uses: actions/checkout@v4
+      - name: Remove bin directory to disable developer mode
+        run: rm -rf ./bin
       - name: Test GitHub Action
         id: test
         uses: ./
@@ -225,6 +235,8 @@ jobs:
           docker load --input /tmp/asyncapi.tar
           docker image ls -a
       - uses: actions/checkout@v4
+      - name: Remove bin directory to disable developer mode
+        run: rm -rf ./bin
       - name: Test GitHub Action
         id: test
         uses: ./
@@ -257,6 +269,8 @@ jobs:
           docker load --input /tmp/asyncapi.tar
           docker image ls -a
       - uses: actions/checkout@v4
+      - name: Remove bin directory to disable developer mode
+        run: rm -rf ./bin
       - name: Test GitHub Action
         uses: ./
         with:
@@ -291,6 +305,8 @@ jobs:
           docker load --input /tmp/asyncapi.tar
           docker image ls -a
       - uses: actions/checkout@v4
+      - name: Remove bin directory to disable developer mode
+        run: rm -rf ./bin
       - name: Make output directory
         run: mkdir -p ./output/bundle
       - name: Test GitHub Action
@@ -325,6 +341,8 @@ jobs:
           docker load --input /tmp/asyncapi.tar
           docker image ls -a
       - uses: actions/checkout@v4
+      - name: Remove bin directory to disable developer mode
+        run: rm -rf ./bin
       - name: Test GitHub Action
         uses: ./
         with:

--- a/github-action/entrypoint.sh
+++ b/github-action/entrypoint.sh
@@ -30,10 +30,12 @@ echo "::group::Debug information"
 if [ -f "$workdir/bin/run" ]; then
   echo -e "${BLUE}Running in developer environment...${NC}"
   echo -e "${BLUE}Setting alias for asyncapi:${NC}" "$workdir/bin/run"
+  shopt -s expand_aliases
   alias asyncapi="$workdir/bin/run"
 elif [ -f "$workdir/../bin/run" ]; then
   echo -e "${BLUE}Running in developer environment...${NC}"
   echo -e "${BLUE}Setting alias for asyncapi:${NC}" "$workdir/../bin/run"
+  shopt -s expand_aliases
   alias asyncapi="$workdir/../bin/run"
 elif [ -n "$CLI_VERSION" ] && [ ! "$CLI_VERSION" == "latest" ]; then
   echo -e "${BLUE}CLI version:${NC}" "$CLI_VERSION"

--- a/github-action/entrypoint.sh
+++ b/github-action/entrypoint.sh
@@ -26,7 +26,16 @@ PARAMETERS="$7"
 CUSTOM_COMMAND="$8"
 
 echo "::group::Debug information"
-if [ -n "$CLI_VERSION" ] && [ ! "$CLI_VERSION" == "latest" ]; then
+# Check if currently running in Developer environment by checking for presence of ./bin/run or ../bin/run and set alias asyncapi= PATH_TO_BIN/RUN
+if [ -f "$workdir/bin/run" ]; then
+  echo -e "${BLUE}Running in developer environment...${NC}"
+  echo -e "${BLUE}Setting alias for asyncapi:${NC}" "$workdir/bin/run"
+  alias asyncapi="$workdir/bin/run"
+elif [ -f "$workdir/../bin/run" ]; then
+  echo -e "${BLUE}Running in developer environment...${NC}"
+  echo -e "${BLUE}Setting alias for asyncapi:${NC}" "$workdir/../bin/run"
+  alias asyncapi="$workdir/../bin/run"
+elif [ -n "$CLI_VERSION" ] && [ ! "$CLI_VERSION" == "latest" ]; then
   echo -e "${BLUE}CLI version:${NC}" "$CLI_VERSION"
   # Check if the CLI version is already installed or not
   if [ -z $(command -v -- asyncapi) ]; then

--- a/package.json
+++ b/package.json
@@ -174,7 +174,7 @@
     "createhook": "oclif generate hook myhook --event=command_not_found",
     "createhookinit": "oclif generate hook inithook --event=init",
     "action:docker:build": "docker build -f github-action/Dockerfile -t asyncapi/github-action-for-cli:latest .",
-    "action:test": "cd github-action && make test"
+    "action:test": "npm run build && cd github-action && make test"
   },
   "types": "lib/index.d.ts"
 }

